# Importing Libraries

In [19]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns
%matplotlib inline

## Reading Data

In [20]:
data=pd.read_csv("titanic_dataset.csv") 
data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

## Setting PassengerId as index

In [21]:
data=data.set_index('PassengerId')

## Identifying Data

In [22]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 891 entries, 1 to 891
Data columns (total 11 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  891 non-null    int64  
 1   Pclass    891 non-null    int64  
 2   Name      891 non-null    object 
 3   Sex       891 non-null    object 
 4   Age       714 non-null    float64
 5   SibSp     891 non-null    int64  
 6   Parch     891 non-null    int64  
 7   Ticket    891 non-null    object 
 8   Fare      891 non-null    float64
 9   Cabin     204 non-null    object 
 10  Embarked  889 non-null    object 
dtypes: float64(2), int64(4), object(5)
memory usage: 83.5+ KB


## Shape of Data 
#### (This means dataset has 891 records and 11 columns)

In [23]:
data.shape

(891, 11)

## Finding out columns with missing entries

In [24]:
data.isnull().sum()

Survived      0
Pclass        0
Name          0
Sex           0
Age         177
SibSp         0
Parch         0
Ticket        0
Fare          0
Cabin       687
Embarked      2
dtype: int64

Using a data dictionary dataframe to examine the dataset

In [25]:
datadict = pd.DataFrame(data.dtypes) #Display Datatypes
datadict = datadict.rename(columns={0:'DataType'}) # Renaming column to DataType
datadict['MissingVal'] = data.isnull().sum() #Display Number of Missing Values
datadict['NUnique']=data.nunique() #Display Number of Unique Values
datadict['Count']=data.count() #Display the Count for each column
datadict

DataType  MissingVal  NUnique  Count
Survived    int64           0        2    891
Pclass      int64           0        3    891
Name       object           0      891    891
Sex        object           0        2    891
Age       float64         177       88    714
SibSp       int64           0        7    891
Parch       int64           0        7    891
Ticket     object           0      681    891
Fare      float64           0      248    891
Cabin      object         687      147    204
Embarked   object           2        3    889

## Describing Object Datatypes

In [26]:
data.describe(include=['object'])

Name   Sex  Ticket    Cabin Embarked
count                       891   891     891      204      889
unique                      891     2     681      147        3
top     Braund, Mr. Owen Harris  male  347082  B96 B98        S
freq                          1   577       7        4      644

## Describing Numerical Datatypes

In [27]:
data.describe(include=['number'])

Survived      Pclass         Age       SibSp       Parch        Fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200